lib/slosilo/adapters/sequel_adapter.rb

```diff
@@ -50,13 +50,17 @@ def each
       end
 
       def recalculate_fingerprints
-        model.each do |m|
-          m.update fingerprint: Slosilo::Key.new(m.key).fingerprint
+        # Ensures all updates are successfull, otherwise it reverts the updates
+        model.db.transaction do
+          model.each do |m|
+            m.update fingerprint: Slosilo::Key.new(m.key).fingerprint
+          end
         end
       end
 
 
       def migrate!
+        warn("`migrate!` is deprecated, please use `recalculate_fingerprints`")
```

Comment:
After looking at this file before we made any changes, I think we should leave this deprecation warning out of this PR.

While I don't think that `migrate!` matters to anyone but us, `recalculate_fingerprints` is also not a replacement for this migration. Rather, it simply factors out the fingerprint calculation to be able to use just that on its own.

`migrate!` checks whether the fingerprint column exists in the database, and actually does migrate the database to include it if not.

We probably can safely remove this altogether, but let's leave it out of the scope of this PR, because the deprecation and replacement is not actually correct.